### PR TITLE
Add validation for zip file name

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-807

### Change description ###
- Add validation for zip filename to verify if it is matching with the zip filename in metadata.
- Update metadata test files

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchMetaDataException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=4998895&bid=16496448)

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchMetaDataException.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.exceptions;
+
+public class ZipNameNotMatchMetaDataException extends InvalidEnvelopeException {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchMetaDataException.java]: either 'not match**ing**' or '**does** not match', I think.

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeProcessorValidationTest.java [hunk 2]
```diff
@@ -266,6 +268,22 @@ public void should_throw_an_exception_when_jurisdiction_poBox_and_container_dont
...
+
+        // when
+        Throwable throwable = catchThrowable(
+            () -> assertZipFilenameMatchesWithMetadata(envelope, "invalid-zip-filename.zip")
+        );
+
+        // then
+        assertThat(throwable)
+            .isInstanceOf(ZipNameNotMatchMetaDataException.class)
+            .hasMessage("Zip file name uploaded does not match with field \"zip_file_name\" in the metadata");
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeProcessorValidationTest.java]: how about "Name of uploaded zip file...."?

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchingMetaDataException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=4998895&bid=16496448)

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeProcessorValidationTest.java (hunk 2 above)
Comment (aliveni) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeProcessorValidationTest.java]: updated the error message

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchMetaDataException.java (hunk 1 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchMetaDataException.java]: renamed as `ZipNameNotMatchingMetaDataException`

**Comment by doncem** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchingMetaDataException.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/bulk-scan-processor/pullRequest?prid=4998895&bid=16496448)

**Pull request merged**